script.js

```diff
@@ -0,0 +1,360 @@
+
+const peopleArray = [
+  {
+    name: 'Adam',
+    age: 20
+  },
+  {
+    name: 'Amanda',
+    age: 5
+  },
+  {
+    name: 'John',
+    age: 75
+  },
+  {
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getAgeDifference(peopleArray)); // 70
+
+function getYoungestPerson(people) {
+    const userSortedByAge = people.slice().sort(
+        function (firstUser, secondUser) {
+            return firstUser.age - secondUser.age;
+        }
+    )
+    return userSortedByAge[0];
+}
+
+function getOldestPerson(people) {
+    const userSortedByAge = people.slice().sort(
+        function (firstUser, secondUser) {
+            return firstUser.age - secondUser.age;
+        }
+    )
+    return userSortedByAge[userSortedByAge.length - 1];
+}
+
+function getAgeDifference(people) {
+    const youngestPerson = getYoungestPerson(people);
+    const oldestPerson = getOldestPerson(people);
+    return oldestPerson.age - youngestPerson.age;
+}
+
+// 8. You get an array of numbers, return the sum of all of the positives ones.
+// Example [1,-4,7,12] => 1 + 7 + 12 = 20
+// Note: if there is nothing to sum, the sum is default to 0.
+
+function getPositiveSum(arrayOfNumbers) {
+    let sumOfNumbers = 0;
+    for (let i = 0; i < arrayOfNumbers.length; i++) {
+        if (arrayOfNumbers[i] > 0) {
+            sumOfNumbers = sumOfNumbers + arrayOfNumbers[i];
+        }
+    }
+    return sumOfNumbers;
+}
+
+console.log(getPositiveSum([1,-4,7,12]));
+
+// 9. Complete the square sum function so that it squares each number passed into it and then sums the results together.//
+// For example, for [1, 2, 2] it should return 9.
+
+function getSquareSum(arrayOfNumbers) {
+    let squaredSum = 0;
+    for (let i = 0; i < arrayOfNumbers.length; i++) {
+        const squaredNumber = arrayOfNumbers[i] ** 2;
+        squaredSum = squaredSum + squaredNumber;
+    }
+    return squaredSum;
+}
+
+console.log(getSquareSum([1, 2, 2]));
+
+// 10. Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.
+// The output should be two capital letters with a dot separating them.
+// It should look like this:
+// Sam Harris => S.H
+// patrick feeney => P.F
+
+function getInitials(name) {
+    const splitName = name.split(' ');
+    let initials = splitName[0].substring(0, 1).toUpperCase();
+    if (splitName.length > 1) {
+        initials += '.' + splitName[splitName.length - 1].substring(0, 1).toUpperCase();
+    }
+    return initials;
+}
+
+// Example usage
+console.log(getInitials("Sam Harris")); // Output: "S.H"
+console.log(getInitials("patrick feeney")); // Output: "P.F"
+
+// 11. Consider an array/list of sheep where some sheep may be missing from their place.
+// We need a function that counts the number of sheep present in the array (true means present).
+
+function getCountSheeps(sheep) {
+    let numberOfSheeps = 0;
```

Comment:
In English, the plural form of "sheep" is still "sheep".
The same for words like fish - there is no word "fishes".

We have some words like that in Polish as well, such as "drzwi".